Remove uWSGI / WSGI support

mvdbeek opened the pull request:
Addresses https://github.com/galaxyproject/galaxy/issues/12257

This removes support for running Galaxy as a WSGI application (so removes both uWSGI and paster support).
To continue testing interactive tools this PR adds the possibility to run tests where Galaxy is driven by gravity (https://github.com/galaxyproject/galaxy/pull/13071/commits/945ecca943fd5a3627c0d708afbec2a1c54acce7)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

mvdbeek committed:
WIP remove paster/uwsgi

mvdbeek committed:
First pass in dropping uwsgi specific things

mvdbeek committed:
Drop tests for WEB_APP=dev

mvdbeek committed:
Make sure SERVER_PORT is string before passing on to hgwebdir

mvdbeek committed:
Drive tool shed tests with uvicorn

mvdbeek committed:
Drop more uWSGI legacy things

mvdbeek committed:
Fix get_ip on OSX

mvdbeek committed:
Raise exception when waiting for entry point

There's a retry loop, I suppose it was meant to be used that way.

mvdbeek committed:
Run interactivetool tests against a gravity config

mvdbeek committed:
Drop uwsgi section from unit test test config files

mvdbeek committed:
Debug failure to start in GravityWrapper

nsoranzo commented on lib/galaxy/containers/docker_swarm.py:
```diff
@@ -125,7 +125,7 @@ def run_in_container(self, command, image=None, **kwopts):
         if self._conf.managed and self._conf.manager_autostart:
             try:
-                # sys.exectuable would be preferable to using $PATH, but sys.executable is probably uwsgi
+                # sys.exectuable would be preferable to using $PATH, but sys.executable is probably gunicorn
```
```suggestion
                # sys.executable would be preferable to using $PATH, but sys.executable is probably gunicorn
```

nsoranzo replied:
Is this comment actually still valid? We could probably replace `"python"` with `sys.executable` below.

mvdbeek replied:
We discussed dropping the whole package (I think in the testing WG meeting), I don't think this is seeing any use. I'd probably do that in another PR though

nsoranzo commented on lib/galaxy/util/web_compat.py:
```diff
@@ -1,6 +1,6 @@
 """Work around for gross circular dependency between galaxy.util and galaxy.web_stack.
```
This file can removed as it's not needed any more after commit 1497b31f0409e5de1f60dc5480a702925580860b .

nsoranzo commented on test/unit/config/config_manage/test_config_manage.py:
```diff
@@ -21,10 +21,6 @@ def test_reports_conversion_1607_sample():
         reports_config = config["reports"] or {}
         assert "use_beaker_session" not in reports_config
```
```suggestion
        assert "use_beaker_session" not in reports_config
        assert "uwsgi" not in config
```

nsoranzo commented on test/unit/app/jobs/test_job_configuration.py:
```diff
@@ -82,19 +73,10 @@ def application_stack(self) -> ApplicationStack:
-                "galaxy.web_stack.uwsgi.opt", self._uwsgi_opt
-            ), mock.patch("galaxy.jobs.JobConfiguration.DEFAULT_BASE_HANDLER_POOLS", base_handler_pools):
-                self._job_configuration = JobConfiguration(self.app)
+            self._job_configuration_base_pools or JobConfiguration.DEFAULT_BASE_HANDLER_POOLS
```
Is this line doing anything?

mvdbeek replied:
Doesn't look like it, I'll drop it

nsoranzo commented on run.sh:
```diff
@@ -50,9 +50,6 @@ if [ "$INITIALIZE_TOOL_DEPENDENCIES" -eq 1 ]; then
 fi
 
-[ -n "$GALAXY_UWSGI" ] && APP_WEBSERVER='uwsgi'
-find_server "${GALAXY_CONFIG_FILE:-none}" galaxy
```
I think only the first of these 2 lines should be deleted, the other is needed to set the `run_server` variable.

nsoranzo commented on run.sh:
```diff
@@ -50,9 +50,6 @@ if [ "$INITIALIZE_TOOL_DEPENDENCIES" -eq 1 ]; then
-[ -n "$GALAXY_UWSGI" ] && APP_WEBSERVER='uwsgi'
-find_server "${GALAXY_CONFIG_FILE:-none}" galaxy
-
 if [ "$run_server" = "python" -a -n "$GALAXY_RUN_ALL" ]; then
```
`$run_server` cannot be `python` any more, so the `if` branch here can be dropped.

nsoranzo commented on lib/galaxy/config/config_manage.py:
```diff
@@ -396,7 +184,7 @@ class App(NamedTuple):
     destination: str
     schema_path: str
-    uwsgi_module: str
+    app_factory: str
```
It seems that this attribute is not used, drop?

mvdbeek committed:
Drop unused app_factory

mvdbeek committed:
Drop galaxy_vscode.ini

It's long been possible to use galaxy.yml for running galaxy inside of
vscode.

mvdbeek committed:
Drop legacy app_factory fallback

mvdbeek committed:
Drop web_compat.py

mvdbeek committed:
Clean up run.sh

mvdbeek committed:
Clean up unit tests

mvdbeek committed:
Drop init scripts that are likely unused

mvdbeek committed:
Drop "starting galaxy" docs

They're outdated and changing rapidly. Without context I don't think
they're all that helpful.

mvdbeek committed:
Fix tool shed tests

nsoranzo commented on lib/galaxy/config/config_manage.py:
```diff
@@ -674,12 +421,6 @@ def _run_conversion(args, app_desc):
                 _warn(message)
                 continue
```
I think everything from `server_section = None` till here can be removed.

nsoranzo commented on lib/galaxy/config/config_manage.py:
```diff
@@ -674,12 +421,6 @@ def _run_conversion(args, app_desc):
-
     app_items = _find_app_options_from_config_parser(p)
     applied_filters = []
     if filters:
```
We can delete also from here...

nsoranzo commented on lib/galaxy/config/config_manage.py:
```diff
@@ -691,8 +432,6 @@ def _run_conversion(args, app_desc):
                     _warn(f"Unknown filter found [{value}], exiting...")
                     sys.exit(1)
```
... to here.

nsoranzo committed:
Fix typo

nsoranzo commented on lib/galaxy/web_stack/__init__.py:
```diff
@@ -220,372 +185,12 @@ def set_postfork_server_name(self, app):
-
-class PasteApplicationStack(ApplicationStack):
-    name = "Python Paste"
+class WebApplicationStack(ApplicationStack):
```
Unused?

mvdbeek replied:
Should be used by non-gunicorn asgi application servers (running my dev instance with uvicorn for instance), but I'll double check

nsoranzo commented on lib/galaxy/web_stack/__init__.py:
```diff
@@ -717,25 +315,5 @@ def get_app_kwds(config_section, app_name=None):
-
-if uwsgi:
-    uwsgi.post_fork_hook = _do_uwsgi_postfork
+def get_stack_facts(config=None):
```
Also unused?

nsoranzo commented on lib/galaxy/webapps/galaxy/buildapp.py:
```diff
@@ -57,7 +57,7 @@ def app_pair(global_conf, load_app_kwds=None, wsgi_preflight=True, **kwargs):
     else:
         try:
-            app = galaxy.app.UniverseApplication(global_conf=global_conf, **kwargs)
+            app = galaxy.app.UniverseApplication(global_conf=global_conf, application_stack="fastAPI", **kwargs)
```
The new `application_stack` also doesn't seem to be used?

mvdbeek committed:
Drop unused get_stack_facts

mvdbeek committed:
Drop unused application_stack argument

mvdbeek committed:
Fall back to generic WebApplicationStack

if we're building a web app, but aren't using gunicorn. I don't think
this has a big impact currently, since you can only set base pools via
lib/galaxy/main.py. I think you could set up alternative logging based
on the web stack class though, and this seems cleaner.

mvdbeek committed:
Drop converting things for which we have no equivalent
